### PR TITLE
FIX: Make docker-compose run server and database correctly.

### PR DESCRIPTION
**Pull request by cesarila**
After making these changes, I can run the db and server with:
```
docker-compose build
docker-compose up
```

### CODE REVIEW

**Comment by cesarila** [comment]
I also changed to using a named volume for the database during this, since my previous implementation was creating a dangling database volume on every run. If you need to recreate the database from the dump for any reason, do the following:
```bash
docker-compose down #Just to make sure the containers are removed
docker volume rm portal2-cm-boards-v2_postgres_data
docker-compose up
```

**Comment by DanielBatesJ** [comment]
I don't see why the folder structure for the server needs to be changed to support using volumes rather than copying over the directory. What is the conflict with a more traditional volume mapping like the one from this [version](https://github.com/p2sr/portal2-cm-boards-v2/blob/67c50cea432c23650dae86953e732bde7c9171f8/docker-compose.yaml)?
```
volumes: 
    - ./server:/server
```
When testing docker previously this achieved what I wanted without any extra issues, is there something that prevents it being this simple and requires the new directory structure?

**Comment by cesarila** [comment]
> I don't see why the folder structure for the server needs to be changed to support using volumes rather than copying over the directory. What is the conflict with a more traditional volume mapping like the one from this [version](https://github.com/p2sr/portal2-cm-boards-v2/blob/67c50cea432c23650dae86953e732bde7c9171f8/docker-compose.yaml)?
> 
> ```
> volumes: 
>     - ./server:/server
> ```
> 
> When testing docker previously this achieved what I wanted without any extra issues, is there something that prevents it being this simple and requires the new directory structure?

The entrypoint script needs have the executable bit set on the host machine if we were to use the previous directory structure. The reason I did it the way I did was so that I could copy the script over to the container and set its executable bit there.

The bind-mounted volume is not available at build time, so the only way around it that I know of to guarantee the script is executable is to copy the script in from somewhere and then set the executable bit at build time. It made the most sense to change the folder structure to me, but I am open to other solutions.

It may be possible to keep the folder structure the same as before and copy the entrypoint script to somewhere other than where the bind-mounted volume ends up and set its executable bit there at build time. I fear that it is much less clear that the container needs to be rebuilt upon changes to the entrypoint script with that approach though.

**Comment by DanielBatesJ** [comment]
I think if the solution is a little less convenient, but we don't have a strange folder directory that could confuse contributors it would work better. The docker implementation is supposed to be mostly set and forget, if it makes it more confusing to navigate I think it loses some of its appeal.

File: docker-compose.yaml [hunk 1]
```diff
@@ -4,8 +4,6 @@ services:
     container_name: "web-server"
     build: ./server
     command: cargo watch -x 'run'
-    depends_on:
-      - postgres
```
Comment (cesarila) [inline docker-compose.yaml:12]: I was running an outdated version of docker-compose (v1.29.2), so depends_on was doing less than it does with v2. After updating to v2, I removed the depends_on entirely to be able to do some cargo stuff in the background while waiting for the database to be ready. This only has a meaningful effect in cases where the postgres_data volume needs to be rebuilt, and only costs 0.07 seconds on my machine in cases where the postgres_data volume does not need to be rebuilt.

**Comment by DanielBatesJ** [comment]
I have added more details to the README, mostly unrelated to your specific instructions (I only changed formatting). I added information about adding the `.env` as it seems that in the current iteration, it uses the .env that's mounted (I can't tell if the others are just being ignored now?). But details about what needs to be changed in docker vs local development and which version users should build are up now.

I think after maybe checking to see if `.env.server.dev` can be removed, we should be able to merge this to master.

**Comment by cesarila2** [comment]
Yep. I think your changes to the README allow for the removal of .env.server.dev. I'll follow the .env directions after deleting .env.server.dev and make sure everything still works. If all is good, I'll commit without .env.server.dev.

**Pull request merged**